### PR TITLE
feat(Message): [modal] when duration set 0, the message will not turn off automatically

### PR DESCRIPTION
**Pull request by AcWrong02**
# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-vue/blob/dev/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?
当duration设置为0时，message不会自动关闭

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
  - Added a new button to trigger a warning message in the modal that does not automatically close, requiring manual closure by the user.

- **Improvements**
  - Enhanced descriptions for the `defaultValue` field and delay settings in both Chinese and English, providing clearer information about automatic shutdown settings.

- **Bug Fixes**
  - Updated modal behavior to set timers conditionally, ensuring timers are only set if the duration is not '0' or a non-zero number.

### CODE REVIEW

File: packages/renderless/src/modal/index.ts [hunk 1]
```diff
@@ -269,12 +273,14 @@ export const open =
       if (state.isMsg) {
         api.addMsgQueue()
 
-        timer = window.setTimeout(
-          () => {
-            api.close(params.type)
-          },
-          parseFloat(props.duration as string)
-        )
+        if (props.duration !== '0') {
```
Comment (James-9696) [inline packages/renderless/src/modal/index.ts]: @AcWrong02 是否考虑: duration：number | string 两种类型？

Comment (AcWrong02) [reply packages/renderless/src/modal/index.ts]: > @AcWrong02 是否考虑: duration：number | string 两种类型？

感谢review，是要为number|string的

File: packages/renderless/src/modal/index.ts [hunk 2]
```diff
@@ -157,20 +157,24 @@ export const selfClickEvent =
...
   (): void => {
     api.addMsgQueue()
 
-    timer = window.setTimeout(
-      () => {
-        api.close('close')
-      },
-      parseFloat(props.duration as string)
-    )
+    if (props.duration !== '0') {
```
Comment (zzcr) [inline packages/renderless/src/modal/index.ts]: 这里没有判断是数字number为0的情况吧？

Comment (AcWrong02) [reply packages/renderless/src/modal/index.ts]: 新增上去了

File: packages/renderless/src/modal/index.ts [hunk 3]
```diff
@@ -269,12 +273,14 @@ export const open =
       if (state.isMsg) {
         api.addMsgQueue()
 
-        timer = window.setTimeout(
-          () => {
-            api.close(params.type)
-          },
-          parseFloat(props.duration as string)
-        )
+        if (props.duration !== '0' || (typeof props.duration === 'number' && props.duration !== 0)) {
```
Comment (kagol) [inline packages/renderless/src/modal/index.ts:276]: @AcWrong02 这个判断逻辑似乎有问题，当 `props.duration` 为 0 时，会命中 `props.duration !== '0'` 这个条件，从而执行关闭逻辑，这实际上不是预期的行为，预期的行为是 `props.duration` 为 0 时，message 不应该关闭才对。

**Pull request closed without merging**